### PR TITLE
Atac seq

### LINKED ISSUE #50: New module : ATAC-seq
New ATAC-seq module from @mirax87 is allready there as PR : #49 .

We should merge it once the new Chip-Seq_revamp is merged (I still have to open the PR) I am waiting for Felix Kruger (SNPsplit author) to fix one issue with SNPsplit, after which it would be fine.

After merging Chip-seq revamp, we would tag **v0.5** , then I after merging ATAC-Seq we would tag **v0.6** ..

### PR DESCRIPTION
**Pull request by mirax87**
Major changes in ATAC-seq workflows with focus on peak calling only (closes #50 )
1) ATAC-seq now depends on DNA-mapping
2) Added QC metrics
3) No tag-shifting (not critical for peak calling)
4) Read filtering of fragments greater x nt ( x := 150 [default])
5) MACS2 peak calling in BEDPE mode

### CODE REVIEW

**Comment by mirax87** [comment]
Initially, I planned a MAPQ filter for the ATAC-seq reads. Currently, the filter in ATAC-seq is missing. However, it reads can be filtered by the preceeding DNA-mapping workflow

**Comment by mirax87** [comment]
FrIP is never cmputed for ATAC-seq. Required to be fixed before merge.

File: shared/rules/ATAC.snakefile [hunk 1]
```diff
@@ -1,3 +1,10 @@
+chromlist = {}
```
[the remark this answers is not in the document]
Comment (mirax87) [reply shared/rules/ATAC.snakefile]: At this point _chromlist_ is a white list for selectiong the reads used in peak calling. 

I plan to replace it by our blacklist approach.

[the remark this answers is not in the document]
Comment (vivekbhr) [reply shared/rules/ATAC.snakefile]: What I mean is you can inherit this directly from the ignorForNorm param in yaml. no need to hard code it inside the snakefile

File: shared/rules/ATAC.snakefile [hunk 2]
```diff
@@ -8,59 +15,66 @@
...
+        os.path.join(outdir_MACS2, "{sample}.all.bedpe")
+    params:
+        samtools_macs2filter="-f 2 -F 4 -F 8 -F 256 -F 512 -F 2048"
     shell:
-            "/package/bedtools2/bin/bedtools bamtobed -bedpe -i {input} | cut -f 1,2,6 | "
-            "awk -v OFS='\\t' -v pos_offset=\"4\" -v neg_offset=\"5\" "
-            "'{{ if( $9 == \"+\" ) {{ print($1, $2 + pos_offset , $3 - neg_offset ) }} "
-            "else {{print($1, $2 - neg_offset , $3 + pos_offset) }} }}' > {output}"
+        samtools_path + "samtools view -b {params.samtools_macs2filter} {input} |"
+        "/package/bedtools2/bin/bedtools bamtobed -bedpe -i - | "
```
Comment (vivekbhr) [inline shared/rules/ATAC.snakefile]: add bedtools_path

File: workflows/ATAC-seq/example.atac_samples.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+################################################################################
+# Please specify all ChIP samples plus their matching control/chromatin input
+# sample.
+# Specify for each ChIP sample whether the IP target results in broad/mixed-type
```
Comment (vivekbhr) [inline workflows/ATAC-seq/example.atac_samples.yaml]: Can you update this as per ATAC-Seq?

File: shared/rules/deepTools_ATAC.snakefile [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        bai = "filtered_bam/{sample}.filtered.bam"
+    output:
+        os.path.join(deeptools_ATAC, "{sample}.fragments.bw" )
+    params:
+        bw_binsize = bw_binsize,
+        genome_size = genome_size,
+        read_extension = "--extendReads"
+    threads: 16
+    shell:
+        deepTools_path+"bamCoverage "
```
[the remark this answers is not in the document]
Comment (mirax87) [reply shared/rules/deepTools_ATAC.snakefile]: Resolved by removal (DNA-mapping produces these tracks)

File: shared/rules/deepTools_ATAC.snakefile [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        png = "--plotFile " +os.path.join(deeptools_ATAC ,
+         "plotFingerprint","plotFingerprint.png") if (len(samples)<=20)
+              else ""
+    log:
+        os.path.join(deeptools_ATAC, "logs","plotFingerprint.log")
+    benchmark:
+        os.path.join(deeptools_ATAC, ".benchmark","plotFingerprint.benchmark")
+    threads: 24
+    shell:
+        deepTools_path+"plotFingerprint "
```
[the remark this answers is not in the document]
Comment (mirax87) [reply shared/rules/deepTools_ATAC.snakefile]: Resolved. Yet, untested. Please try with your new data set

File: workflows/ATAC-seq/example.atac_samples.yaml (hunk 3 above)
Comment (mirax87) [reply workflows/ATAC-seq/example.atac_samples.yaml]: Resolved

**Comment by vivekbhr** [comment]
@mirax87 you had defined the function `merge_dicts` once in the wrapper but don't use it.. i removed it in the last PR.. is it ok?

**Comment by mirax87** [comment]
@vivekbhr Perfectly fine. Seems like `merge_dicts` was an artifact from earlier development.

**Comment by vivekbhr** [comment]
1. --properpairs is required for the DNA mapping before ATAC-seq
2. `deeptools_cmds()` were not imported in the ATAC-seq snakefile.

**Comment by vivekbhr** [comment]
@mirax87 can you please check whether you have updated the workflow on your end? If yes, please push the new changes and I will then merge this into develop..

At this point the workflow successfully completes.

**Comment by mirax87** [comment]
Ready to merge. Removing WIP tag

Latest changes consist of 
- Error message, if bedtools bamtobed fails 
- Removed sample config
- Code cleaned

**Pull request merged**